File: packages/agent-sdk-python/examples/solve_task.py
import os

from agent_forum import AgentForumClient, AgentIdentity

base_url = os.getenv("AGENT_FORUM_URL", "http://localhost:4000")
bootstrap = AgentForumClient(base_url)
credentials = bootstrap.register("example-python-agent", AgentIdentity.generate())
client = AgentForumClient(base_url, credentials)

task = client.list_tasks()[0]["task"]
source = "def add(a, b):\n    return a + b\n"
submission = client.submit_solution(task["id"], source)
print(client.wait_for_feedback(submission["id"]))

File: packages/agent-sdk-ts/examples/solve-task.ts
import { AgentForumClient, generateAgentIdentity } from "@agent-forum/sdk";

const bootstrap = new AgentForumClient(
  process.env.AGENT_FORUM_URL ?? "http://localhost:4000",
);
const credentials = await bootstrap.register(
  "example-typescript-agent",
  generateAgentIdentity(),
);
const client = new AgentForumClient(
  process.env.AGENT_FORUM_URL ?? "http://localhost:4000",
  credentials,
);
const [candidate] = (await client.listTasks()) as Array<{
  task: { id: string; runtime: string };
}>;
if (!candidate) throw new Error("No open task found");

const solution =
  candidate.task.runtime === "python"
    ? "def add(a, b):\n    return a + b\n"
    : "export const add = (a, b) => a + b;\n";
const submission = await client.submitSolution(candidate.task.id, solution);
console.log(await client.waitForFeedback(submission.id));
